system-addon/lib/TelemetryFeed.jsm

```diff
@@ -296,11 +296,27 @@ this.TelemetryFeed = class TelemetryFeed {
   }
 
   createUserEvent(action) {
-    return Object.assign(
+    let pingCentrePing = Object.assign(
       this.createPing(au.getPortIdOfSender(action)),
       action.data,
       {action: "activity_stream_user_event"}
     );
+
+    let utExtras = this.createPing(au.getPortIdOfSender(action));
+    delete utExtras.locale;
+    for (let field of Object.keys(utExtras)) {
+      utExtras[field] = String(utExtras[field]);
+    }
+
+    let mainFields = ["event", "source", "action_position"];
+    let eventFields = mainFields.map(field => String(pingCentrePing[field]) || null);
+    let eventsTelemetryPing = [
+      "activity_stream",
+      ...eventFields,
+      utExtras
+    ];
+
+    return {pingCentrePing, eventsTelemetryPing};
```

Comment:
This will attempt to turn *any* user event ping into an events telemetry ping but at the moment it should only send `CLICK` pings. This should instead only create `pingCentrePing` and exit early if it's not a `CLICK` ping.

Reply:
The schema definition (https://bugzilla.mozilla.org/show_bug.cgi?id=1429497) has been updated to include **all** event types not just `CLICK` so this is no longer an issue.